packages/api/tests/models/test-executions-model.js

```diff
@@ -1,344 +1,344 @@
+  t.false(Object.keys(actualParams.ExpressionAttributeValues).includes(':wrong'));
+  t.false(actualParams.UpdateExpression.includes('wrong'));
 });
 
-test('Updating an existing record updates the record as expected', async (t) => {
-  const { arn, executionName } = t.context;
+test('buildDocClientUpdateParams() returns the correct result for a running item', (t) => {
+  const { executionModel } = t.context;
 
-  const message = createExecutionMessage({ executionName });
+  const item = {
+    arn: 'abc-123',
+    status: 'running',
+    createdAt: 123,
+    updatedAt: 124,
+    timestamp: 124,
+    originalPayload: 'my-original-payload'
+  };
 
-  await executionModel.createExecutionFromSns(message);
-  const originalRecord = await executionModel.get({ arn });
+  const actualParams = executionModel.buildDocClientUpdateParams(item);
 
-  const finalPayload = { test: 'payloadValue' };
-  message.meta.status = 'completed';
-  message.payload = finalPayload;
+  t.is(actualParams.TableName, process.env.ExecutionsTable);
+  t.deepEqual(actualParams.Key, { arn: 'abc-123' });
 
-  await executionModel.updateExecutionFromSns(message);
+  t.true(actualParams.UpdateExpression.startsWith('SET '));
+  t.false(actualParams.UpdateExpression.includes('REMOVE '));
+  t.false(actualParams.UpdateExpression.includes('ADD '));
+  t.false(actualParams.UpdateExpression.includes('DELETE '));
 
-  const record = await executionModel.get({ arn });
+  t.is(actualParams.ExpressionAttributeNames['#status'], 'status');
+  t.is(actualParams.ExpressionAttributeValues[':status'], 'running');
+  t.true(actualParams.UpdateExpression.includes('#status = if_not_exists(#status, :status)'));
 
-  const expectedRecord = {
-    ...originalRecord,
-    finalPayload,
-    status: 'completed',
-    type: 'test',
-    duration: record.duration,
-    timestamp: record.timestamp,
-    updatedAt: record.updatedAt
+  t.is(actualParams.ExpressionAttributeNames['#createdAt'], 'createdAt');
+  t.is(actualParams.ExpressionAttributeValues[':createdAt'], 123);
+  t.true(actualParams.UpdateExpression.includes('#createdAt = :createdAt'));
+
+  t.is(actualParams.ExpressionAttributeNames['#updatedAt'], 'updatedAt');
+  t.is(actualParams.ExpressionAttributeValues[':updatedAt'], 124);
+  t.true(actualParams.UpdateExpression.includes('#updatedAt = :updatedAt'));
+
+  t.is(actualParams.ExpressionAttributeNames['#timestamp'], 'timestamp');
+  t.is(actualParams.ExpressionAttributeValues[':timestamp'], 124);
+  t.true(actualParams.UpdateExpression.includes('#timestamp = :timestamp'));
+
+  t.is(actualParams.ExpressionAttributeNames['#originalPayload'], 'originalPayload');
+  t.is(actualParams.ExpressionAttributeValues[':originalPayload'], 'my-original-payload');
+  t.true(actualParams.UpdateExpression.includes('#originalPayload = :originalPayload'));
+});
+
+test('buildDocClientUpdateParams() always updates values for a non-running item', (t) => {
```

Comment:
We're only testing the status?

Reply:
Talked about this in person. We could check any field, so I chose status.